src/instruction.rs

```diff
@@ -0,0 +1,185 @@
+//! Program instructions
+
+use solana_program::clock::UnixTimestamp;
+
+/// Instructions supported by the staking program
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum StakeInstruction {
+    /// Creates Stake config account which controls staking parameters
+    ///
+    /// 0. `[writable]` Stake config account
+    /// 1. `[]` Slash authority, may be a PDA for a governance program
+    /// 2. `[]` Config authority, may be a PDA for a governance program
```

Comment:
Do these two need to be passed in as accounts, or can they be keys in the input data?

Reply:
They don't need to be passed in as accounts, but it can be better since you potentially dedupe the pubkey when compiling the instruction, and also get free validation that it's a well-formed pubkey. I lean towards doing it this way, but I can be swayed

Reply:
Since initialize is only called once on a config, I don't think slightly higher compute usage should be a problem in exchange for some more clarity and safety.